project/src/main/puzzle/piece/piece-queue.gd

```diff
@@ -245,10 +245,12 @@ func _insert_annoying_piece(max_pieces_to_right: int) -> void:
 		Utils.remove_all(extra_piece_types, pieces[new_piece_index - 1].type)
 		if new_piece_index < pieces.size():
 			Utils.remove_all(extra_piece_types, pieces[new_piece_index].type)
-	if extra_piece_types[0] == PieceTypes.piece_o:
-		# the o piece is awful, so it comes 10% less often
-		extra_piece_types.shuffle()
-	pieces.insert(new_piece_index, _new_next_piece(extra_piece_types[0]))
+	
+	if extra_piece_types:
+		if extra_piece_types[0] == PieceTypes.piece_o:
+			# the o piece is awful, so it comes 10% less often
+			extra_piece_types.shuffle()
+		pieces.insert(new_piece_index, _new_next_piece(extra_piece_types[0]))
```

Comment:
Worth mentioning this in the commit comment.

Reply:
Added comment comment, `Fixed array out of bounds if you made a level with four identical pieces
like ['O', 'V', 'O', 'V']`